tests/test_gcloud.py

```diff
@@ -0,0 +1,264 @@
+    from pyramid_storage import gcloud
+    g = gcloud.GCloudFileStorage(
+        credentials="/secrets/credentials.json",
+        bucket_name="my_bucket",
+        extensions="images"
+    )
+    assert not g.extension_allowed("jpg", ("gif",))
+
+
+def test_extension_not_allowed_if_allowed_if_not_dotted():
+    from pyramid_storage import gcloud
+    g = gcloud.GCloudFileStorage(
+        credentials="/secrets/credentials.json",
+        bucket_name="my_bucket",
+        extensions="images"
+    )
+    assert not g.extension_allowed("jpg", ("gif",))
+
+
+def test_file_allowed():
+    from pyramid_storage import gcloud
+
+    fs = mock.Mock()
+    fs.filename = "test.jpg"
+    s = gcloud.GCloudFileStorage(
+        credentials="/secrets/credentials.json",
+        bucket_name="my_bucket",
+        extensions="images"
+    )
+
+    assert s.file_allowed(fs)
+
+
+def test_file_not_allowed():
+    from pyramid_storage import gcloud
+
+    fs = mock.Mock()
+    fs.filename = "test.jpg"
+    g = gcloud.GCloudFileStorage(
+        credentials="/secrets/credentials.json",
+        bucket_name="my_bucket",
+        extensions="documents"
+    )
+
+    assert not g.file_allowed(fs)
+
+
+def test_save_if_file_not_allowed():
+    from pyramid_storage import gcloud
+    from pyramid_storage.exceptions import FileNotAllowed
+
+    fs = mock.Mock()
+    fs.filename = "test.jpg"
+    g = gcloud.GCloudFileStorage(
+        credentials="/secrets/credentials.json",
+        bucket_name="my_bucket",
+        extensions="documents"
+    )
+
+    with pytest.raises(FileNotAllowed):
+        g.save(fs)
+
+
+def test_save_if_file_allowed():
+    from pyramid_storage import gcloud
+
+    fs = mock.Mock()
+    fs.filename = "test.jpg"
+
+    g = gcloud.GCloudFileStorage(
+        credentials="/secrets/credentials.json",
+        bucket_name="my_bucket",
+        extensions="images")
+
+    with mock.patch(
+            'pyramid_storage.gcloud.GCloudFileStorage.get_connection',
+            _get_mock_gcloud_connection):
+        name = g.save(fs)
+    assert name == "test.jpg"
+
+
+def test_save_file():
+    from pyramid_storage import gcloud
+
+    g = gcloud.GCloudFileStorage(
+        credentials="/secrets/credentials.json",
+        bucket_name="my_bucket",
+        extensions="images"
+    )
+
+    with mock.patch(
+            'pyramid_storage.gcloud.GCloudFileStorage.get_connection',
+            _get_mock_gcloud_connection):
+        name = g.save_file(mock.Mock(), "test.jpg")
+    assert name == "test.jpg"
+
+
+def test_save_filename():
+    from pyramid_storage import gcloud
+
+    g = gcloud.GCloudFileStorage(
+        credentials="/secrets/credentials.json",
+        bucket_name="my_bucket",
+        extensions="images"
+    )
+
+    patches = (
+        mock.patch(_mock_open_name(), _mock_open),
+        mock.patch(
+            'pyramid_storage.gcloud.GCloudFileStorage.get_connection',
+            _get_mock_gcloud_connection
+        )
+    )
```

Comment:
If we mock this in each test, we should do a TestCase with a `setUp()`

Reply:
Actually adding a testcase mean adding unittest to the mix. It landed in Python 3* but since this lib is supporting up to python2.6 I am a bit reluctant starting this.

The code in test_s3 also call the patch multiple times. Can we delegate this in the refactor test issue to come?